torch/csrc/jit/passes/onnx/fixup_onnx_loop.cpp

```diff
@@ -68,8 +70,172 @@ void FixupONNXLoops(Block* block) {
   }
 }
 
+namespace {
+bool IsErasableSequence(const Node* loop_node, size_t i) {
+  AT_ASSERT(loop_node->blocks().size() == 1);
+  auto* sub_block = loop_node->blocks()[0];
+  auto* out_node = sub_block->outputs()[i-1]->node();
+  auto* in_val = sub_block->inputs()[i];
+
+  if (out_node->kind() != ::c10::onnx::SequenceInsert) {
+    return false;
+  }
+
+  if (out_node->inputs().size() == 3) {
+    // Non-default insert position is not supported.
+    return false;
+  }
+
+  if (out_node->input(0) != in_val) {
+    // Only SequenceInsert that applies on loop-carried sequence is supported.
+    return false;
+  }
+
+  const auto seq_node_kind = loop_node->inputs()[i]->node()->kind();
+  if (seq_node_kind != ::c10::onnx::SequenceEmpty) {
+    // Initial sequence must be empty.
```

Comment:
I'm just curious why the sequence has to be empty?

Reply:
nvm

Reply:
The way we support append in loop is by converting the tensor list to scan output in ONNX loop. That requires the intial tensor list to be empty.